test/it/optimizer/src/test/java/org/apache/shardingsphere/test/it/optimize/SQLNodeConverterEngineIT.java

```diff
@@ -193,6 +193,9 @@ private Set<String> getSupportedSQLCaseIDs() {
             result.add("select_order_by_for_nulls_last");
             result.add("select_char");
             result.add("select_weight_string");
+            result.add("select_substring");
```

Comment:
In addition, I recommend supporting only one new SQL CASE at a time, which can help solve the problem.

Reply:
Okay, I will :)